data/cherrytree.1
.TH CHERRYTREE "1" "September 2022" "cherrytree 0.99.49"
.SH NAME
cherrytree \- a hierarchical note taking application
.SH SYNOPSIS
\fBcherrytree [\-V] [\-N] [filepath [\-n nodename] [\-a anchorname] [\-x export_to_html_dir] [\-t export_to_txt_dir] [\-p export_to_pdf_path] [\-P password] [\-w] [\-s]]\fP
.SH DESCRIPTION
\fBcherrytree\fP is a hierarchical note taking application, featuring rich
text, syntax highlighting, images handling, hyperlinks, import/export with
support for multiple formats, support for multiple languages, and more.
.SH AUTHOR
\fBcherrytree\fP was written by Giuseppe Penone <giuspen@gmail.com> and Evgenii Gurianov <https://github.com/txe>.
.PP
This manual page was written by Vincent Cheng <Vincentc1208@gmail.com>,
for the Debian project (and may be used by others).


changelog

0.28.4 (November 12th, 2012)
• fixed internationalization/localization bugs and other minor bugs

0.28.3 (October 31st, 2012)
• updated French language: welcome to Benoît

0.28.2 (October 31st, 2012)
• fixed bugs: cannot save file as in 0.28.1

0.28.1 (October 31st, 2012)
• fixed several bugs
• ctrl+period inserts a newline in a table cell

0.28 (October 10th, 2012)
• reduced the size of the preferences dialog adopting tabs
• some special symbols are automatically inserted after typing the characters that compose them followed by spacebar; ctrl+z undoes the replacement ( --> to → , <-- to ← , <-> to ↔ , (c) to © , (tm) to ™ , (r) to ® )
• added import from Mempad
• fixed several bugs

0.27.1 (September 23rd, 2012)
• fixed bug: print / save as pdf of the whole tree not working in 0.27

0.27 (September 18th, 2012)
• added preference option to enable the periodic check of the cherrytree document file for external updates
• added preference option to choose between appindicator and systray for panel docking
• added opportunity to restore the latest search all matches dialog with shortcut (Ctrl+Shift+A)
• added preference options for text and tree to choose between dark theme, light theme or custom colors
• added the opportunity to print / save as pdf / export of the selected text only
• added the opportunity to include the node name in print / save as pdf
• added import from Gnote (code from Rahul Sundaram)
• shortcut for paste as plain text in rich text node (Ctrl+Shift+V)
• adopted python ConfigParser for saving preferences, old preferences will be lost
• fixed several bugs
• a new French language maintainer is needed, thanks Ludovic for the great job you made

0.26.3 (July 4th, 2012)
• fixed bugs

0.26.2 (June 11th, 2012)
• fixed bugs

0.26.1 (June 6th, 2012 )
• added language Chinese Simplified (great job from Channing Wong)
• fixed minor bug

0.26 (May 16th, 2012)
• added text formatting superscript, subscript, monospace and h3
• h1, h2 and h3 formatting is now automatically applied to the whole row
• the switch between rich text and automatic syntax highlighting for the nodes is now simpler
• the dialog to choose the file storage type is now file extension oriented
• it is now possible to configure the default rich text foreground and background, so is for the tree
• the horizontal rule inserted characters are now configurable
• the TOC now includes h3 and can be created also for the whole tree
• the links anchors size is now configurable
• fixed several bugs
• we have now a complete manual of the application linked at help–manual (great job from Robert)
• WARNING: documents edited with a version >= 0.26 cannot be edited again with a version < 0.26

0.25.4 (February 24th, 2012)
• fixed bugs

0.25.3 (February 3rd, 2012)
• fixed bugs

0.25.2 (January 24th, 2012)
• fixed bugs

0.25.1 (January 21st, 2012)
• fixed bugs
• mouse middle-click (or wheel-click) on a tree node expands/collapses it

0.25 (January 14th, 2012)
• on linux, cherrytree can now be installed on any “prefix”, but the “exec-prefix” is not to be set different than “prefix”
• the minimum required gtk version is now 2.16
• the single (but very big) new feature is a new data storage type, a sqlite 3 database (which is anyway a single file). Storing the old way (a single xml file) will be still available but probably not much used since the new storage type offers faster reading, faster saving and less memory usage. To the two old file extensions “ctd” and “ctz” (xml and xml 7zipped+password protected) were added “ctb” and “ctx” (sqlite and sqlite 7zipped+password protected)
• since really a lot of code changed I ask everybody to backup the data and report possible bugs. To try the new storage type just click file–save as

0.24 (December 28th, 2011)
• automatic syntax highlighting: it is now possible to choose between several style schemes (a couple of them with dark background), the same style scheme is used for tree nodes and codeboxes
• automatic syntax highlighting: added the options to show white spaces and highlight the current line
• added menu items and keyboard shortcuts to move the current row/selected rows up and down the document
• added support for appindicator where the system tray is not supported (ubuntu unity)
• fixed several bugs and bad behaviours

0.23.1 (September 16th, 2011)
• fixed bugs

0.23 (September 4th, 2011)
• the limit of undoable steps per node is now configurable (lower the limit to save ram)
• implemented a dialog to iterate graphical way the find/replace (before it was possible only through keyboard shortcuts)
• implemented the insertion of timestamps. the timestamp format is configurable through preferences dialog
• fixed some bugs/bad behaviors related to system tray, nodes drag n drop, links to files dialog and others

0.22.2 (July 10th, 2011)
• fixed bugs

0.22.1 (June 23rd, 2011)
• bugs fixed

0.22 (June 7th, 2011)
• fixed bugs
• added the right click option ‘paste as plain text’ to the rich text nodes
• added imports from tomboy and knowit
• all the exports (pdf/html/txt/ctd) now offer the options: only current node / current node and subnodes / all the tree
• implemented simple clickable to-do lists

0.21.3 (May 3rd, 2011)
• updated German language
• fixed minor bugs

0.21.2 (April 29th, 2011)
• fixed minor bugs

0.21.1 (April 26th, 2011)
• fixed critical bug (involving first cherrytree installation only, not upgrades and not 0.21 windows version)

0.21 (April 23rd, 2011)
• fixed bugs
• added links to folders
• added custom action for clicked link to files and folders
• added some customizable properties to the codeboxes
• added fullscreen mode
• implemented the cut/copy/paste of a table row
• tables: pressing arrows up/down switches to former/next cell to edit, pressing return also switches to the next cell to edit
• now multiple instances are allowed and handled
• added the opportunity to sort the whole tree (before it was possible to sort only the siblings of the selected node)

scripts/homebrew/cherrytree.rb
class Cherrytree < Formula
  desc "Hierarchical note taking application featuring rich text and syntax highlighting"
  homepage "https://www.giuspen.net/cherrytree/"
  url "https://www.giuspen.net/software/cherrytree_0.99.38.tar.xz"
  sha256 "788a64566b14b0f390111cbadc50c6dcb410c4dac90cabffbf0a191f14a46b60"
  license "GPL-3.0-or-later"

  livecheck do
    url :homepage
    regex(/href=.*?cherrytree[._-]v?(\d+(?:\.\d+)+)\.t/i)
  end

  bottle do
    sha256 arm64_big_sur: "7ab37082920cc336b2dec0380eba94cb0c0d777f095357cf7e5cf32452cfd603"
    sha256 big_sur:       "03b521744828c72049eaa6da515f2d231c31f264a39f6e5fe573e678c0394c7b"
    sha256 catalina:      "944dfa361bc1dd35775f1ce5fa523edfcfcaedbfd50148671cbb03f6e4102c5d"
    sha256 mojave:        "b646c2dc2c0ee5f1a1140a5746e47a27b9a8e36f7ba3b367c30858ac78b165ca"
  end

  depends_on "cmake" => :build
  depends_on "ninja" => :build
  depends_on "pkg-config" => :build
  depends_on "python@3.9" => :build
  depends_on "adwaita-icon-theme"
  depends_on "fmt"
  depends_on "gspell"
  depends_on "gtksourceviewmm3"
  depends_on "libxml++"
  depends_on "spdlog"
  depends_on "uchardet"

  uses_from_macos "curl"

  def install
    system "cmake", ".", "-DBUILD_TESTING=''", "-GNinja", *std_cmake_args
    system "ninja"
    system "ninja", "install"
  end

  test do
    (testpath/"homebrew.ctd").write <<~EOS
      <?xml version="1.0" encoding="UTF-8"?>
      <cherrytree>
        <bookmarks list=""/>
        <node name="rich text" unique_id="1" prog_lang="custom-colors" tags="" readonly="0" custom_icon_id="0" is_bold="0" foreground="" ts_creation="1611952177" ts_lastsave="1611952932">
          <rich_text>this is a </rich_text>
          <rich_text weight="heavy">simple</rich_text>
          <rich_text> </rich_text>
          <rich_text foreground="#ffff00000000">command line</rich_text>
          <rich_text> </rich_text>
          <rich_text style="italic">test</rich_text>
          <rich_text> </rich_text>
          <rich_text family="monospace">for</rich_text>
          <rich_text> </rich_text>
          <rich_text link="webs https://brew.sh/">homebrew</rich_text>
        </node>
        <node name="code" unique_id="2" prog_lang="python3" tags="" readonly="0" custom_icon_id="0" is_bold="0" foreground="" ts_creation="1611952391" ts_lastsave="1611952667">
          <rich_text>print('hello world')</rich_text>
        </node>
      </cherrytree>
    EOS
    system "#{bin}/cherrytree", testpath/"homebrew.ctd", "--export_to_txt_dir", testpath, "--export_single_file"
    assert_predicate testpath/"homebrew.ctd.txt", :exist?
    assert_match "rich text", (testpath/"homebrew.ctd.txt").read
    assert_match "this is a simple command line test for homebrew", (testpath/"homebrew.ctd.txt").read
    assert_match "code", (testpath/"homebrew.ctd.txt").read
    assert_match "print('hello world')", (testpath/"homebrew.ctd.txt").read
  end
end
